File: ThreatModelDfdService/Data/DTO/UpsertDfdElementDTO.cs
using ThreatModelDfdService.Model.Enums;

namespace ThreatModelDfdService.Data.DTO;

public record UpsertDfdElementDTO (
    long? Id,
    string Name,
    DfdElementType Type,
    decimal XValue,
    decimal YValue,
    decimal Width,
    decimal Height,
    string Uuid
);

File: dfd-service/src/main/java/com/tcc/dfd_service/dto/UpsertDfdElementDTO.java
package com.tcc.dfd_service.dto;

import com.tcc.dfd_service.enums.DfdElementType;
import java.math.BigDecimal;

public record UpsertDfdElementDTO(
        Long id,
        String name,
        DfdElementType type,
        BigDecimal xValue,
        BigDecimal yValue,
        BigDecimal width,
        BigDecimal height,
        String uuid
) {
}